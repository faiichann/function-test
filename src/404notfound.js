import React,{Component} from 'react';
import {Redirect} from 'react-router-dom';
import styled from 'styled-components';

export const Button = styled.button`
  /* Adapt the colors based on primary prop */
  /**ส่งprops เชคว่าเป็น primary หรือไม่ให้แสดงผล ถ้าไม่ :  */
  background: ${props => props.primary ? "palevioletred" : "white"}; 
  color: ${props => props.primary ? "white" : "palevioletred"};

  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  cursor:pointer;
 
`

// const Button2 =  styled(Button)`
//   &:hover{
//     background-color: #000;
//     color:#fff;
//   }

// `

class NotFoundPage extends Component{
    state={
        Redirect:false
      }
    render(){
        if (this.state.Redirect){
            return <Redirect to='/'/>
          }
        return(
            <>
            <h1>404 Page Not Found</h1>
            <Button  onClick={()=>this.setState({Redirect:true})}>Home</Button>
            </>
        )
    }
}

export default NotFoundPage