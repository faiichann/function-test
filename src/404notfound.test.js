import React from 'react';
import NotFoundPage from './404notfound'
import { shallow,mount } from 'enzyme';
import {Button} from './404notfound'
import renderer from 'react-test-renderer';
import 'jest-styled-components';

it('renders', () => {
  const wrapper = shallow(<NotFoundPage />);
  expect(wrapper.find('h1').text()).toEqual('404 Page Not Found');
});

it('Button style',() =>{
const wrapper = mount(<Button primary onClick={jest.fn()}/>);
const tree = renderer.create(wrapper).toJSON();
// expect(tree).toMatchSnapshot()
expect(tree).toHaveStyleRule('background','palevioletred')
});

it('button click', () => {
  const setState = jest.fn()
  const wrapper = shallow(<NotFoundPage  />);
  const button = wrapper.find(Button)
  button.simulate('click')
  expect(wrapper.state('Redirect')).toBe(true);
  // expect(setState).toHaveBeenCalledTimes(0)
});
