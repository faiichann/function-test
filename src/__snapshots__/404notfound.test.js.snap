// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button style 1`] = `
Array [
  <h1>
    404 Page Not Found
  </h1>,
  .c0 {
  background: white;
  color: palevioletred;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  cursor: pointer;
}

<button
    className="c0"
    onClick={[Function]}
  >
    Home
  </button>,
]
`;
